docs/tasks.md

```diff
@@ -6,105 +6,82 @@ weight: 1
 -->
 # Tasks
 
-A `Task` (or a [`ClusterTask`](#clustertask)) is a collection of sequential
-steps you would want to run as part of your continuous integration flow. A task
-will run inside a pod on your cluster.
-
-A `Task` declares:
-
-- [Parameters](#parameters)
-- [Resources](#resources)
-- [Steps](#steps)
-- [Workspaces](#workspaces)
-- [Results](#results)
-
-A `Task` is available within a namespace, and `ClusterTask` is available across
-entire Kubernetes cluster.
-
----
-
-- [ClusterTasks](#clustertask)
-- [Syntax](#syntax)
-  - [Steps](#steps)
-    - [Step script](#step-script)
-  - [Parameters](#parameters)
-  - [Resources](#resources)
-    - [Inputs](#inputs)
-    - [Outputs](#outputs)
-  - [Workspaces](#workspaces)
-  - [Results](#results)
-  - [Volumes](#volumes)
-  - [Step Template](#step-template)
-  - [Sidecars](#sidecars)
-  - [Variable Substitution](#variable-substitution)
-- [Examples](#examples)
-- [Debugging Tips](#debugging)
-
-## ClusterTask
-
-Similar to Task, but with a cluster scope.
-
-In case of using a ClusterTask, the `TaskRef` kind should be added. The default
-kind is Task which represents a namespaced Task
-
-```yaml
-apiVersion: tekton.dev/v1beta1
-kind: Pipeline
-metadata:
-  name: demo-pipeline
-  namespace: default
-spec:
-  tasks:
-    - name: build-skaffold-web
-      taskRef:
-        name: build-push
-        kind: ClusterTask
-      params: ....
-```
-
-A `Task` functions exactly like a `ClusterTask`, and as such all references to
-`Task` below are also describing `ClusterTask`.
-
-## Syntax
-
-To define a configuration file for a `Task` resource, you can specify the
-following fields:
+- [Overview](#overview)
+- [Configuring a `Task`](#configuring-a-task)
+  - [`Task` vs. `ClusterTask`](#task-vs-clustertask]
+  - [Defining `Steps`](#defining-steps)
+    - [Running scripts within `Steps`](#running-scripts-within-steps)
+  - [Specifying `Parameters`](#specifying-parameters)
+  - [Specifying `Resources`](#specifying-resources)
+    - [Specifying inputs](#specifying-inputs)
+    - [Specifying outputs](#specifying-outputs)
+  - [Specifying `Workspaces`](#configuring-workspaces)
+  - [Storing execution results](#storing-execution-results)
+  - [Specifying `Volumes`](#configuring-volumes)
+  - [Specifying a `Step` template](#specifying-a-step-template)
+  - [Using `Sidecars`](#using-sidecars)
```

Comment:
i think this link needs to be updated too - im gonna stop clicking on all of them now and just recommend trying out all the links :D